coreneuron/logging/mpi_communicator_impl.hpp

```diff
@@ -0,0 +1,165 @@
+#include "coreneuron/mpi_communicator.hpp"
+
+#include "coreneuron/logging.hpp"
+
+
+namespace coreneuron
+{
+  namespace mpi
```

Comment:
As in many other places in this PR it's not entirely clear where your choice comes from and what this particular change is there for. There is a lot of documentation / commenting missing